dex/elrond_dex_farm/src/lib.rs

```diff
@@ -161,43 +162,43 @@ pub trait Farm:
         let burn_gas_limit = self.burn_tokens_gas_limit().get();
 
         let farm_attributes = self.get_farm_attributes(&payment_token_id, token_nonce)?;
-        let mut enter_amount = &farm_attributes.total_entering_amount * &liquidity
-            / farm_attributes.total_liquidity_amount.clone();
-        require!(enter_amount > 0, "Cannot exit farm with 0 entering amount");
-
-        // For locked rewards option, exit before minumum epochs is not allowed
         require!(
             !farm_attributes.with_locked_rewards
                 || !self.should_apply_penalty(farm_attributes.entering_epoch),
             "Exit too early for lock rewards option"
         );
 
-        // Before removing liquidity, first generate the rewards.
         let mut reward_token_id = self.reward_token_id().get();
-        self.increase_actual_reserves(&self.mint_rewards(&reward_token_id));
+        self.generate_aggregated_rewards(&reward_token_id);
 
-        let caller = self.blockchain().get_caller();
-        let mut reward = self.remove_liquidity(&liquidity, &enter_amount)?;
-        let farming_token_id = self.farming_token_id().get();
-        self.send()
-            .burn_tokens(&farm_token_id, token_nonce, &liquidity, burn_gas_limit);
+        let mut reward = self.calculate_reward(
+            &amount,
+            &self.reward_per_share().get(),
```

Comment:
is reward_per_shard derived from rewards_per_block and total_liquidity ?

Reply:
reward per share is an accumulator. at each state it adds the rewards that were generated divided by current total liquidity